### PR TITLE
add missing test mocks

### PR DESCRIPTION
**Pull request by mjaskula**
This PR fixes a handful of test related issues that have been plaguing recent PRs:

- Add the `rp2` module to the test mocks to allow the PolySquare script to be added to the menu (PR #141)
- add the `framebuf` module to the test mocks to allow the cvrecorder script to use it (removed in PR #139)
- update the europi lib with an alternate method for detecting the micropython environment (issue #124), this allows the cvrecorder to use the `micropython` module (removed in PR #121).

### CODE REVIEW

**Comment by mjaskula** [comment]
@gamecat69 @t-schreibs This PR should solve test issues in your recent PRs.

File: software/contrib/cvecorder.py [hunk 1]
```diff
@@ -86,7 +87,7 @@ def __init__(self):
 
         # Test routine, pick a random bank n times and save, then load the state
         if self.initTest:
-            #print(micropython.mem_info("level"))
+            print(micropython.mem_info("level"))
```
Comment (awonak) [inline software/contrib/cvecorder.py:90]: Let's remove or comment this so we don't unnecessarily waste cycles when the modules isn't connected to a PC.

File: software/contrib/cvecorder.py [hunk 2]
```diff
@@ -271,7 +272,7 @@ def saveState(self):
                     print(f'[{attempts}] Error: Memory allocation failed, retrying: {e}')
                 if self.debugLogging:
                     self.writeToDebugLog(f"[saveState] Error: Memory allocation failed, retrying: {e}")
-                    #print(micropython.mem_info("level"))
+                    print(micropython.mem_info("level"))
```
Comment (awonak) [inline software/contrib/cvecorder.py:275]: Same here

**Review by awonak — APPROVED** [review]

File: software/contrib/cvecorder.py (hunk 1 above)
Comment (mjaskula) [reply software/contrib/cvecorder.py:90]: This is behind the `initTest` flag, which needs to be manually enabled by editing the code. Presumable the user knows what they are doing if they are running this test.

File: software/contrib/cvecorder.py (hunk 2 above)
Comment (mjaskula) [reply software/contrib/cvecorder.py:275]: Same thing here, this is behind the `debugLogging` which is manually enabled.

File: software/contrib/cvecorder.py (hunk 1 above)
Comment (awonak) [reply software/contrib/cvecorder.py:90]: I missed that it was guarded by a flag. Thanks for clarifying. This addresses my concern.

**Pull request merged**